### ISSUE TITLE
bug: Rating unchecked displays all as checked

### ISSUE THREAD

**Issue by pkpg** [saadeghi/daisyui#3308]
### What version of daisyUI are you using?

v4.12.20

### Which browsers are you seeing the problem on?

All browsers

### Reproduction URL

https://play.tailwindcss.com/KFg2Be57Wl

### Describe your issue

If none of the stars is checked, all stars display as checked.

**Comment by cluk3** [comment]
Hey @saadeghi looks like the problem is the following css in src/components/styled/rating.css:

```
 input:checked ~ input,
  input[aria-checked="true"] ~ input {
    @apply bg-opacity-20;
  }
```

so when there isn't any input checked the `bg-opacity-20` will not apply. I propose to invert the logic to have `bg-opacity-20` as default and then apply full opacity to the checked input and the one before:

```
input:checked, input:has(~ input:checked) {
  @apply bg-opacity-100;
}
```

I can open a PR myself if you wish!

**Comment by saadeghi** [comment]
@cluk3 that's right. I already fixed that in v5 alpha.  https://github.com/saadeghi/daisyui/blob/112ea1a47e4ff1155cc7be99a3262a4b8e614fdc/packages/daisyui/src/components/rating.css#L28

I will also do a patch tomorrow to fix this in v4.
